pkg/runtime/wfengine/actor_backend.go

```diff
@@ -110,24 +117,41 @@ func getActivityScheduler(activityWorkItemChan chan *backend.ActivityWorkItem) a
 }
 
 // InternalActors returns a map of internal actors that are used to implement workflows
-func (be *actorBackend) GetInternalActorsMap() map[string]actors.InternalActor {
+func (abe *ActorBackend) GetInternalActorsMap() map[string]actors.InternalActor {
 	internalActors := make(map[string]actors.InternalActor)
-	internalActors[be.config.workflowActorType] = be.workflowActor
-	internalActors[be.config.activityActorType] = be.activityActor
+	internalActors[abe.config.workflowActorType] = abe.workflowActor
+	internalActors[abe.config.activityActorType] = abe.activityActor
 	return internalActors
 }
 
-func (be *actorBackend) SetActorRuntime(actors actors.Actors) {
-	be.actors = actors
+func (abe *ActorBackend) SetActorRuntime(actorRuntime actors.ActorRuntime, ctx context.Context) {
+	abe.actors = actorRuntime
+	abe.actorRuntime = actorRuntime
+	if abe.actorsReady.CompareAndSwap(false, true) {
```

Comment:
Are we anticipating that multiple goroutines might call this concurrently? If so, should we also only assign the `actors` and `actorRuntime` fields within this block?

Reply:
I see that this method was copied over from another file. The reason for the CAS was to be able to call `close()` on the channel, which is used to send a signal that the workflow engine has set an actor runtime to everyone that was waiting. See: https://github.com/dapr/dapr/pull/6972